utils/proxy/proxyconfig.go

```diff
@@ -163,6 +163,9 @@ var portMap = map[string]string{
 func canonicalAddr(url *url.URL) string {
 	addr := url.Host
 	if !hasPort(addr) {
+		if strings.HasPrefix(addr, "[") && strings.HasSuffix(addr, "]") {
```

Comment:
The square brackets thing is for URLs. JoinHostPort does not deal with URLs, it deals with addresses.

The Go 1.8 code for proxies does the right thing and uses URL.Hostname, which strips off the brackets. When we move everything over to Go 1.8, we should look to drop this code. (I forget why we have it in the first place?)